### PR TITLE
Fix memory usage of VarVerletListsAsBuild

### LINKED ISSUE #322: VarVerletListsAsBuild uses way too much memory
**Describe the bug**
The `VarVerletListsAsBuild` container uses way to much memory with data layout SoA and many threads.

**To Reproduce**
1. Start a simulation with the container, SoA, and many threads.
2. See that the memory usage scales with the number of threads and way to high.

**Expected behavior**
The memory usage should be more or less equal for different numbers of threads.

### PR DESCRIPTION
**Pull request by HobbyProgrammer**
# Description

Fixes the bug that the memory usage of  `VarVerletListsAsBuild` scales with the number of threads used for data layout SoA.

In order to do this, the signature of one SoAFunctor method had to be changed. The changes are minimal, though.

## Resolved Issues

- [x] fixes #322

# How Has This Been Tested?
Unit Tests

### CODE REVIEW

File: src/autopas/pairwiseFunctors/LJFunctor.h [hunk 1]
```diff
@@ -371,25 +371,24 @@ class LJFunctor
 
   // clang-format off
   /**
-   * @copydoc Functor::SoAFunctor(SoAView<SoAArraysType> soa, const std::vector<std::vector<size_t, autopas::AlignedAllocator<size_t>>> &neighborList, size_t iFrom, size_t iTo, bool newton3)
+   * @copydoc Functor::SoAFunctor(SoAView<SoAArraysType> soa, const size_t indexFirst, const std::vector<size_t, autopas::AlignedAllocator<size_t>> &neighborList, bool newton3)
    * @note If you want to parallelize this by openmp, please ensure that there
    * are no dependencies, i.e. introduce colors and specify iFrom and iTo accordingly.
```
Comment (FG-TUM) [inline src/autopas/pairwiseFunctors/LJFunctor.h]: adapt comment

**Comment by FG-TUM** [comment]
@LG-TUM this will affect your functor template :P

**Comment by SteffenSeckler** [comment]
please adapt the comment and resolve the merge conflicts

**Comment by SteffenSeckler** [comment]
merged the master into this PR, will wait on jenkins for merge.

**Pull request merged**